subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/report/listener/LegacyTestArtifactsProcessor.kt

```diff
@@ -0,0 +1,108 @@
+package com.avito.instrumentation.internal.report.listener
+
+import com.avito.android.Result
+import com.avito.report.model.AndroidTest
+import com.avito.report.model.Incident
+import com.avito.report.model.IncidentElement
+import com.avito.report.model.TestRuntimeData
+import com.avito.report.model.TestRuntimeDataPackage
+import com.avito.report.model.TestStaticData
+import com.avito.retrace.ProguardRetracer
+import com.avito.time.TimeProvider
+import com.avito.utils.stackTraceToList
+import com.github.salomonbrys.kotson.fromJson
+import com.google.gson.Gson
+import kotlinx.coroutines.CoroutineName
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+import java.io.File
+import java.io.FileReader
+
+internal class LegacyTestArtifactsProcessor(
+    private val gson: Gson,
+    private val testArtifactsUploader: TestArtifactsUploader,
+    private val retracer: ProguardRetracer,
+    private val timeProvider: TimeProvider
+) : TestArtifactsProcessor {
+
+    override fun process(
+        reportDir: File,
+        testStaticData: TestStaticData,
+        logcatBuffer: LogcatBuffer?
+    ): Result<AndroidTest> {
+
+        val scope = CoroutineScope(CoroutineName("test-artifacts-${testStaticData.name}") + Dispatchers.IO)
+
+        val reportJson = File(reportDir, TestArtifactsProcessor.REPORT_JSON_ARTIFACT)
+
+        return Result.tryCatch {
+            val testRuntimeData: TestRuntimeData = gson.fromJson<TestRuntimeDataPackage>(
+                FileReader(reportJson)
+            )
+
+            val isTestFailed = testRuntimeData.incident != null
+
+            runBlocking {
+                withContext(scope.coroutineContext) {
+                    AndroidTest.Completed.create(
+                        testStaticData = testStaticData,
+                        testRuntimeData = testRuntimeData,
+                        stdout = uploadLogcat(logcatBuffer?.getStdout(), isUploadNeeded = isTestFailed),
```

Comment:
Invoking suspending function. If I'm not mistaken it will be blocking.
In this case we should make it async and upload 2 logs in parallel.

Reply:
fixed